### ISSUE TITLE
get_smac function does not work correctly on virtual interfaces

### ISSUE THREAD

**Issue by cappe987** [cappe987/wiretime#2]
In the file https://github.com/cappe987/wiretime/blob/main/src/liblink.c.

There is a temporary workaround that just sets the MAC to `AA:AA:AA:AA:AA:AA`, but ideally it should get the correct MAC address of the interface being transmitted on.

**Comment by Saavrm26** [comment]
May I work on this issue

**Comment by cappe987** [comment]
> May I work on this issue

Go ahead :)

**Comment by Saavrm26** [comment]
could you provide some resource for creating vlans(linux)
I have the project setup, but i am unable to setup a vlan for testing

**Comment by cappe987** [comment]
Hi @Saavrm26
In trying to write some instructions for you I noticed that this was not an issue anymore. It must've been fixed when I did some refactoring. I'm not really sure myself how it would have occurred in the first place.

If you are still interested in contributing to some of the other issues you are free to do so. There is some explanation for setup in this issue https://github.com/cappe987/wiretime/issues/3. When I said "virtual interface" I was talking about veth interfaces, not VLANs. Veth is basically a virtual port, and are created in pairs that connect (like a cable).
